Allow create and remove functions to operate properly when app_id is …

jaleksynas opened the pull request:
…present and you are using a technical role account rather than admin. Also removed some code duplication.
#1895

jacob-aleksynas committed:
Allow create and remove functions to operate properly when app_id is present and you are using a technical role account rather than admin.

grantjk committed:
Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers

grantjk committed:
Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers

KrauseFx committed:
[match] Fixed spelling mistakes (#4682)

* Fixed spelling mistakes

* Fixed shell escaping

* Fixed spelling mistake

KrauseFx committed:
[spaceship] Updated spaceship gem description (#4684)

Commit:
Don't show private lanes in lanes table (#4690)

bartoszj committed:
Added support for carthage `--derived-data`. (#4163)

dtorres committed:
Added option to specify scheme in update_info_plist instead of plist_path

Added tests for scheme option in update_info_plist

Updated to use Xcodeproj::Plist instead

Commit:
Wrap all Google Api Client calls in error handling in supply (#4681)

KrauseFx committed:
[match] Version bump (#4701)

KrauseFx committed:
Removed link to RubyDoc (#4705)

The documentation there was not up date date

KrauseFx committed:
[match] Updated description git_branch (#4703)

KrauseFx committed:
Replaced remaining .red and raise blocks (#4709)

Replaced `if $verbose` with `UI.verbose`
Removed `Helper.log` method
Removed `Helper.log_alert` method

patoroco committed:
Update rake task name (#4723)

Commit:
Re-implement -itc_provider support for ItunesTransporter

Commit:
Continuing work on #3843

patoroco committed:
Fix gym with path with spaces (#4721)

oysta committed:
Provide an xcversion action to select xcode via version specifier (#4421)

KrauseFx committed:
[fastlane] Updated xcode-install depdendency (#4757)

Commit:
[fastlane_core] Shell escape keychain paths in CertChecker

KrauseFx committed:
[fastlane] Updated platform docs to mention Android by default (#4764)

marcelofabri committed:
Adding more options to pod_lib_lint action (#4771)

KrauseFx committed:
Bartoszj scan devices (#4758)

* Added support for multiple destinations.

KrauseFx committed:
Removed Sentry (#4756)

KrauseFx committed:
[scan] Version bump (#4779)

SiarheiFedartsou committed:
jira action is now verifying 'jira-ruby' gem installed, not 'jira' (#4719)

nomisRev committed:
Added '.' as default value for project_dir in `gradle` action

KrauseFx committed:
[fastlane_core] Prepare for fastlane plugins (#4770)

* [fastlane_core] Prepare for fastlane plugins

- This adds the foundation for 3rd party plugins
- Improved code style

* Added tests

Commit:
Revert "Added support for invocation from actionscript to gradle.rb" (#4789)

KrauseFx committed:
[fastlane_core] Version bump (#4790)

* repare for fastlane plugins (#4770)
* Removed Sentry (#4756)
* Shell escape keychain paths in CertChecker
* Replaced remaining .red and raise blocks (#4709)

KrauseFx committed:
[fastlane] Version bump (#4791)

**New:**
* Provide an `xcversion` action to select Xcode via version specifier (#4421)
* Adding more options to pod_lib_lint action (#4771)
* `scan` can now run scan on multiple devices using the new `devices` option
* Added support for carthage `--derived-data`. (#4163)
* Prepare `fastlane` foundation to support plugins: https://github.com/fastlane/fastlane/issues/4744

**Improvements:**
* Updated platform docs to mention Android by default (#4764)
* Updated `xcode-install` depdendency (#4757)
* Updated internal dependencies
* Replaced remaining .red and raise blocks (#4709)
* Don't show private lanes in lanes table (#4690)
* Updates `Xcake` Action to support  new `xcake make` command. (#4068)
* Carthage configuration option and tvOS platform support (#4054)
* Fix typo in Copy Artifacts (#4645)
* [fastlane] Show custom action message only on verbose mode (#4640)
* jira action is now verifying 'jira-ruby' gem installed
* Fixed escaping of Keychain path
* Removed Sentry crash reporting(#4756)

KrauseFx committed:
[fastlane] Updated dependencies (#4792)

To be merged before https://github.com/fastlane/fastlane/pull/4791

jules2689 committed:
Convert `copy_artifacts` action to Fileutils (#4654)

KrauseFx committed:
[fastlane] Removed open TODOs in tests (#4726)

ravinggenius committed:
Sort team selections (#4794)

* Sort Apple teams alphabetically for easier deterministic scanning

* Memoize Developer Portal team listing response

andybest committed:
Made junit result parsing more robust, to deal with additional properties, orders (#4735)

KrauseFx committed:
[scan] Added documentation for device grid (#4778)

KrauseFx committed:
[fastlane_core] Fixed import check of WWDR certificate (#4802)

KrauseFx committed:
[fastlane_core] Version bump (#4803)

Fixed an issue with importing the WWDR certificate

KrauseFx committed:
[fastlane] Updated fastlane_core dependency (#4806)

KrauseFx committed:
Show message when trying to enable or disable the crash reporter (#4805)

* Show message when trying to enable or disable the crash reporter

* Added direct link to GitHub issues

Fixed white space

fantattitude committed:
Comment for push_to_git_remote was wrong (#4732)

The file was probably copied from add_git_tag.rb and the comment wasn't updated.

mriddle committed:
Pass git_branch to ChangePassord.update (#4808)

* Pass git_branch to ChangePassord.update

This resolves an issue where ChangePassword would set the `branch:` parameter to the value of `params[:branch]`, which would be nil.
Because we were providing the `branch` argument, the default "master" was being ignored. Causing the GitHelper.clone method to explode when trying to clone a nil branch.

* Fix alignment

KrauseFx committed:
[match] Version bump (#4809)

Prepare new release for https://github.com/fastlane/fastlane/pull/4808

znerol committed:
Only query for xcode version on macs (#4631)

bootstraponline committed:
Update slather action with new commands (#4825)

rds13 commented:
👍

Commit:
Scan: Added option to generate the json compilation database with clang
naming conventions

bartoszj committed:
Fixed scan crash on Mac OS X. Link: https://github.com/fastlane/fastlane/pull/4758#discussion-diff-64371251

mfurtak committed:
Version bump

Changes since release 0.7.0:
* Fixed scan crash on Mac OS X. Link: https://github.com/fastlane/fastlane/pull/4758#discussion-diff-64371251
* [scan] Added documentation for device grid (#4778)
* Made junit result parsing more robust, to deal with additional properties, orders (#4735)

Commit:
Re-add Helper.log and add deprecation warning (#4832)

Commit:
fastlane_core version bump (#4833)

Commit:
[snapshot] Improve reset_simulators behavior when makinga phone-watch pair

KrauseFx committed:
Added new `skip_waiting_for_build_processing` option (#4812)

* Added new `skip_waiting_for_build_processing` option

This allows the user to skip the processing time and/or the submission separately

Fixes https://github.com/fastlane/fastlane/issues/1463

* Updated PR to address feedback

KrauseFx committed:
[pilot] Version bump (#4811)

Added new `skip_waiting_for_build_processing` option :+1:

KrauseFx committed:
[fastlane] Updated dependencies (#4835)

KrauseFx committed:
[fastlane] Version bump (#4836)

KrauseFx committed:
[fastlane_core] Added tests for certificate import (#4804)

Fixes https://github.com/fastlane/fastlane/issues/4798 kind of

KrauseFx committed:
[fastlane] Improved handling of new lines in increment_build_number (#4837)

* [fastlane] Improved handling of new lines in increment_build_number

mfurtak committed:
[snapshot] Version bump

Changes since release 1.12.2:
* [snapshot] Improve reset_simulators behavior when makinga phone-watch pair
* [snapshot] Migrated binary to use commands_generator (#4576)

ahmedgharib commented:
I assume this will work with app manager as well?

jaleksynas commented:
The main change (other than some code cleanup and de-duplication) is to use app specific endpoints for tester create/delete.  If a user has the appropriate permissions for the operation in the web UI, this allows the same using pilot.

KrauseFx committed:
[fastlane_core] Version bump (#4869)

Make `determine_version` a class method
Improved unit tests

KrauseFx committed:
Made determine_version a class method (#4870)

KrauseFx committed:
Removed appium development dependency and therefore nokogiri (#4852)

This caused `bundle install` on fastlane to also install nokogiri

KrauseFx committed:
[fastlane] Verify of rest-client dependency for sentry (#4871)

Fixes https://github.com/fastlane/fastlane/issues/4848

Bogidon committed:
Include line number in Artsy link (#4872)

hiltonc committed:
Output changelog in changelog_from_git_commits. (#4860)

mfurtak committed:
[deliver] Version bump

Changes since release 1.11.3:
* Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers

bootstraponline committed:
[slather action] fix --decimals flag type

Fix 'decimals' value must be a String! Found Fixnum instead.

Commit:
Add :dev_portal_team_id option

This helps the iTunes Transporter deal with situations
involving multiple teams.

mfurtak committed:
[pilot] Version bump

Changes since release 1.7.0:
* Add :dev_portal_team_id option

KrauseFx commented on pilot/lib/pilot/manager.rb:
```diff
@@ -56,5 +56,18 @@ def fetch_app_identifier
+    def find_app_id_no_prompt
+      return @apple_id if @apple_id
+      app_filter = (config[:apple_id] || config[:app_identifier])
+      if app_filter
```
This should probably be 
```ruby
return nil unless app_filter
```

KrauseFx committed:
[fastlane_core] Helper.fastlane_enabled? now returns true if .fastlan… (#4889)

* [fastlane_core] Helper.fastlane_enabled? now returns true if .fastlane folder is hidden

KrauseFx committed:
[fastlane] Added syntax highlighting to FAQs (#4891)

KrauseFx committed:
[gym] Print out legacy information every time you run gym (#4890)

Fixes https://github.com/fastlane/fastlane/issues/1866

bgannin committed:
Escape Gradle properties (#4897)

Commit:
[fastlane_core] Fix PkgFileAnalyser to shell escape `xar` arguments

Johennes committed:
Make deliver setup honor the skip_screenshots flag (#4919)

With this commit calling

  deliver init --skip_screenshots

will only download metadata and skip downloading of screenshots. This is
useful for cases where the user is only interested in manipulating and
uploading the metadata as it spares them the lengthy screenshot download.
Before, the only option to skip screenshots during initialization was
to CTRL+C the process after the metadata was downloaded.

ncnlinh committed:
Fix typo in OneSignal action (#4892)

that cause the action to fail every time when creating apps with apns

KrauseFx committed:
Hide timestamp by default (#4838)

* Hide timestamp by default

* Switched to FASTLANE_HIDE_TIMESTAMP instead of show timestamp

KrauseFx committed:
[fastlane] Version bump (#4925)

* Improved handling of new lines in `increment_build_number` (#4837)
* Fix bug in `OneSignal` action (#4892)
* Escape Gradle properties (#4897)
* `slather` fix `--decimals` flag type
* Output changelog in `changelog_from_git_commits` (#4860)
* Verify `rest-client` dependency for `sentry` (#4871)
* Removed `appium` development dependency and therefore `nokogiri` (#4852)

KrauseFx committed:
[frameit] Show full device (#4929)

* Add option to show complete device

For some apps, the bottom of the screen is important. This allows the
developer to specify that this is the case.
Frameit will then shrink the screenshot (if necessary) to ensure that
the entire device and frame is visible in the framed output.

* [frameit] Code style improvements

KrauseFx committed:
[frameit] Version bump (#4930)

Commit:
[spaceship] Add .jpeg to the set of recognized media extensions

KrauseFx committed:
[fastlane] Updated dependencies (#4935)

KrauseFx committed:
[fastlane] Version bump (#4933)

This release adds plugin support to `fastlane`, you can find more information on how to get started at [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

KrauseFx committed:
[Ready] Plugin manager (#4722)

* Added initial Plugins.md

* Added initial PluginManager

* Minor improvements

* Renamed Plugins to Pluginfile

* Added note about storing those files in version control

* Make use of bundler to detect what plugins are currently being loaded

* Fixed spelling mistake

Thanks @jinjorge

* Added tests for PluginManager

* Removed return value for `setup` and `ensure_plugins_attached!` methods

Fixes https://github.com/fastlane/fastlane/pull/4722#discussion_r63905341

* Temporary version bump

* `bundle update` is now no quiet if verbose is on

* Introduced new `available_gems` method

Fixed issue with initial setup

* Added more tests

* Updated and extended docs

* Updated from `fastlane_` to `fastlane-plugin` prefix

* Ask the user for the plugin name if none is provided

* Only ask for user input if no parameter was given

* Fixed tests after recent change

* Added automatic detection if plugin is available on RubyGems and offer user a way to select custom path / url

Added tests

* Added generating of available plugins based on RubyGems API ❤️

* Fixed code style issue

* Added collecting of third party plugins

- Added tracking of third party plugins
- Added detection of reference between plugin and action
- Added tests

* [fastlane_core] Prepare for fastlane plugins

- This adds the foundation for 3rd party plugins
- Improved code style

* Moved fastlane_core tests to other PR

* Added tests

* Added documentation for plugin source

* Fixed plugin manager not using the correct module

* Improved plugin docs

* Skip development dependencies by default when installing plugins

* Removed dead test that didn't actually test things

* Added proper catching of SystemExit and similar to cleanly finish fastlane run

Tested with `xcake`, which throws a `Xcake::Informative` object

* Also track crash when a system exit is used

* Fixed code style

* Removed bundler code from bin/fastlane

* Added more plugins docs

* Added docs for `new_plugin` for existing gems

* Plugin manager (#4776)

* Handle old bundled actions.

* Check on one off as well.

* Update message.

* Added tests for previously bundled fastlane actions

* Fixed spelling mistake in Plugins.md

* Fixed sample path in Plugins.md

* Added test for deprecated actions when plugin is available

* Improved plugin docs

* Updated RubyGems search query

* Added initial Troubleshooting guide

* Improved output when adding a plugin

* Print command when running fastlane install_plugins or fastlane update_plugins

* No more --no-development since it updates the bundler config

* Resolved merge conflict

* Moved all plugin classes into plugins subfolder

* Improved wording and documentation

* Renamed Pluginfile constant

* Changed "pluginsfile_" to "pluginfile_"

* Added plugin summary to see what plugins are loaded

* Improved plugin docs

* Refactored plugin references to also contain version number

Made determine version number a class method with alias to the other method
Added constant for troubleshooting guide

* fastlane now uses plugins itself to run common ruby tasks

* Enabled fastlane plugins for the fastlane's Gemfile

* Updated fastlane_core dependency

* Added support for hidden ".fastlane" folder

* Added new `fastlane search_plugins` command (#4853)

* Added new `fastlane search_plugin` command

* Added tests

* Improved design of search result table

* Switched from `search_plugin` to `search_plugins`

* Removed now: true from tests

* Initial work on plugin generator

* Add more structure and tests around plugin generation

* Add tests to specifically test validation/fixing behavior

* Minimal spec for PluginGeneratoer

* Create spec lets

* One-time initialization in plugin generator spec

* Begin templating files as part of plugin generation

Test the correctness of the gemspec by evaluating it and
checking its values

* Fix hound warnings about exists?

* Fix plugin_generator_ui_spec

* Address feedback, adapt to new prefix, fix paths

* Improve ui object mocking to prevent test hangs if an
  interactive method is not properly mocked
* Fix file generation paths to be correctly start inside
  the gem name directory
* Fix gemspec evaluation during tests to account for its
  new location relative to the temp directory
* Fix plugin info collection to be aware of the changed
  fastlane plugin prefix value
* PluginInfo can generate the require_path from the gem_name
* Command name changed to 'new_plugin'

* Clean up/suppress rubocop violations

* Add email collection to the plugin generator interview

* Pre-fill summary, description, homepage in gemspec; reword info collectory questions

* LICENSE creation

* Add support for collecting gemspec summary and description

* Add missing LICENSE file

* Accept plugin name during command line call

* Define the plugin action class as part of the templating

* Make the plugin gem depend on fastlane

* Make fastlane a development dependency of the plugin bundle

* Add other needed development dependencies to the plugin bundle
  template

* Fix plugin_generator_spec

* Add helper file generation to the plugin generator

* Add .rspec file to the plugin generator content

* Fix failing rubocop offense

* Add Gemfile to plugin generator content

* Add .gitignore to the plugin generator content

* Add main plugin .rb file to the plugin generator content

* Add a success message when the plugin generator is done

* Add spec helper generation

* Add action_spec creation to the plugin generator

* Add Rakefile generation

* Remove description from PluginInfo model

* Add more details the generated README.md

* Continue improving the documentation for plugins and generator

* Update README template for plugin with badge.

* Renamed fastlane logo

* Switched from `capitalize` to `fastlane_class` to work with underlines too

* Show broken plugins in plugins table

* Prepare beta release

* Made bundler a fastlane dependency

* Prepare next beta release

* Optimise and hot badge.

* auto detect email and author from git config

* Suppress command output for git queries during plugin info collection

* Correct incomplete sentence in plugin generator README

* Reimplement the PluginGenerator based on dir templating

Moves all of the plugin generator template files from a flat
directory into one that is actually shaped like the intended
destination structure.

This allows the PluginGenerator to stop encoding knowledge
about the intended structure of the plugin, and just focus
on copying and templating logic.

The template directory uses %-wrapped variables to allow
directory and file names to have variables inserted at
run time. Those variable names are used as accessor calls
into the PluginInfo object to get replacement values.

The tests have been changed to use a plugin name containing
an underscore and have been corrected to check for expected
values based on fastlane_class transform of the plugin name

* Move plugin badge to root fastlane assets dir

Update the plugin README to look in this new path

* Show error message if fastlane directory is missing

* Integrated Club Mate plugin into fastlane setup

* Addressed @mfurtak feedback

* Fixup "Ensure..." line in Gemfile and Pluginfile

* Moved PluginManager object to Fastlane.plugin_manager (#4927)

* Moved PluginManager object to Fastlane.plugin_manager

* Adressed feedback

* Renamed Plugins.md.erb to AvailablePlugins.md.erb

* Adressed PR feedback

* Added minimum requirement for plugins to Plugins.md

* Added install instructions to AvailablePlugins.md

* Fixed plugin generator calling wrong class

* Commented out the spec.homepage

* Updated AvailablePlugins.md

* Moved all plugin specs into subfolder

* Replaced `install_dependencies` with `install_plugins`

* Improved wording

* Minor improvements

* Updated lane name blacklist to not allow plugin commands

* Added auto-import of fastlane actions in plugin specs

KrauseFx committed:
[fastlane_core] Version bump (#4934)

* [fastlane_core] Version bump

Changes since release 0.45.0:
* Hide timestamp by default (#4838)
* [fastlane_core] Fix PkgFileAnalyser to shell escape `xar` arguments
* [fastlane_core] Helper.fastlane_enabled? now returns true if .fastlan… (#4889)
* Continuing work on #3843
* Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers

* [fastlane_core] Version bump

KrauseFx committed:
Enabled fastlane plugins for all the tools (#4941)

KrauseFx committed:
Plugin improvements (#4943)

* Show warning when action is overwritten by plugin

* More Ruby

* Added detection if plugin is already taken on RubyGems

KrauseFx committed:
[fastlane] Updated fastlane_core dependency (#4944)

ohayon committed:
[fastlane] fix changelog_from_git_commits action for cli

ohayon committed:
Fix for CLI based actions with Integer params

KrauseFx committed:
Updated fastlane plugin badge URL (#4948)

mpirri commented:
This looks awesome, @jaleksynas -- can you add some test coverage to this code? Otherwise, we're excited to get this functionality into fastlane!

tcurdt committed:
[testfairy action] enable parameters/options (#4818)

* enable testfairy parameters/options

* make the hound happy

* fixed error message

* validate metrics and options, use symbols

* making the hound happy

* fixed more style issues

jinjorge committed:
Fix broken urls in Spaceship docs (#4952)

neonichu committed:
Add example to plugin template (#4950)

* Add example to plugin template

This allows running `bundle exec fastlane test` directly from the
generated plugin directory and should encourage people to provide an
example `Fastfile`.

* Add missing newlines

* Add generated `fastlane` files to `.gitignore`

* Fix Markdown link

mfurtak committed:
Fix the PluginInfoCollector to check RubyGems for the gem_name

Previously this was checking RubyGems for the plugin_name, which
does not contain the 'fastlane-plugin-' prefix, and is not the
full name of the gem.

Fixes up the tests to match this behavior and provides a helper
method for setting up the stubbed web request.

KrauseFx committed:
Attached Pluginfile to Gemfile for new plugins (#4956)

* Attached Pluginfile to Gemfile for new plugins

* Improved wording for example code

mfurtak committed:
Addressing PR feedback

KrauseFx committed:
[fastlane] Version bump (#4958)

- Improved newly created plugins
  - Fixed badge URL (#4948)
  - Fixed detection of existing plugins (#4957)
  - Added example `Fastfile` and `Pluginfile` to new plugin so it's easy to try out (#4950)
- [testfairy action] enable parameters/options (#4818)

jvanzummeren committed:
Added interline spacing to frameit titles (#3989)

KrauseFx committed:
Print proper error message when new 2step is used (#4959)

Via https://github.com/fastlane/fastlane/issues/4391

KrauseFx committed:
Updated danger dependency

mfurtak committed:
Revert "Allow deliver to work with multiple iTunes Providers" (#4961)

KrauseFx committed:
[fastlane_core] Version bump (#4962)

* Fixed iTunes Transporter
* Fix for CLI based actions with Integer params

Commit:
Updated cocoapods action to correctly update repo when using cocoapods 1.0

cocoapods 1.0 by default does not run pod repo update on pod install
unless --repo-update is set

ohayon committed:
update fastlane core version

ohayon committed:
update danger version

jorgeazevedo committed:
Make CSV header optional in import_testers (#4119)

`pilot import` always skips the first row of a CSV file because it
expects the format

    header1,header2,header3
    column1,column2,column3

whereas the iTunesConnect bulk import function expects a CSV with
format

    column1,column2,column3

This means that if `pilot import` is used with an iTC CSV, it will
skip the first tester in the file.

This patch changes the skipping logic. Instead of skipping the first
row, it skips any row that doesn't have a valid email on the email
column.

By making the header effectively optional, `pilot import` can
successfully process both an iTC CSV and the resulting CSV of `pilot
export`.

mfurtak committed:
[fastlane_core] Version bump

Changes since release 0.46.1:
* Re-implement -itc_provider support for ItunesTransporter
* Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers

Commit:
Add :itc_provider option for deliver and pilot

This provides an explicit option for controlling use of the
-itc_provider option in the iTMSTransporter

mfurtak committed:
[deliver] Version bump

Changes since release 1.12.0:
* Add :itc_provider option for deliver and pilot
* Ads Apple Developer Id to itunes_transporter.rb. Allows deliver to work with multiple iTunes Providers
* Revert "Allow deliver to work with multiple iTunes Providers" (#4961)
* Enabled fastlane plugins for all the tools (#4941)
* Make deliver setup honor the skip_screenshots flag (#4919)

Commit:
Prevent Danger warn when PR is a version bump

mfurtak committed:
[pilot] Version bump

Changes since release 1.8.0:
* Add :itc_provider option for deliver and pilot
* Make CSV header optional in import_testers (#4119)
* Enabled fastlane plugins for all the tools (#4941)

Commit:
Version bump fastlane_core dependency for deliver and pilot

mfurtak committed:
[pilot] Version bump

Changes since release 1.9.0:
* Version bump fastlane_core dependency for deliver and pilot

mfurtak committed:
[deliver] Version bump

Changes since release 1.13.0:
* Version bump fastlane_core dependency for deliver and pilot

Commit:
Continuing work on #4841

pRizz committed:
Improved the accuracy of the hockey description. (#4984)

neerfri committed:
add support for rbenv in wrap_xcodebuild/xcbuild-safe.sh (#4625)

neonichu committed:
Install latest release for an Xcode plugin (#4911)

* Install latest release for an Xcode plugin

Using the `github` option instead of `url`, fastlane can now determine
the URL for the latest version of a plugin automatically.

* Better error handling and minor fixes

- Handles non-existent releases and other GH errors
- Make `:github` option optional
- Fix formatting and use fake URL for `url` in `Actions.md`

KrauseFx committed:
[pilot] Version bump (#4987)

- Stop skipping first line of csv in pilot import

KrauseFx committed:
[danger] Added detection of # in PR-body (#4985)

somedev committed:
Added ability to use certificate SHA-1 in `sigh` resign (#4898)

mfurtak committed:
[scan] Version bump

Changes since release 0.7.1:
* Continuing work on #4841
* Scan: Added option to generate the json compilation database with clang naming conventions
* Enabled fastlane plugins for all the tools (#4941)

KrauseFx committed:
[fastlane] Removed now: true from appium_spec (#4996)

ohayon committed:
Fix issue where user could get stuck in a loop of password prompts

Commit:
Add documentation about itc_provider option for deliver and pilot

ohayon committed:
derp

KrauseFx committed:
Improved plugin default README (#4995)

* Improved plugin default README

* Updated spelling

neonichu committed:
Add `rubocop` to plugin template (#4951)

- Copies shared `.rubocop.yml`
- Add `rake` and `rubocop` as development dependencies
- Add `rspec` and `rubocop` rake tasks
- Fix rubocop issues in the template
- Tests

neonichu committed:
Fix for missing dotfiles in new plugins (#5004)

This happened only when using the packaged gem, because `Dir.glob` does
not include dotfiles by default. Using `FNM_DOTMATCH` changes that
behaviour. I didn't add a test for this, because the error doesn't
surface when developing the gem.

KrauseFx committed:
[fastlane] Version bump (#5002)

* Improved fastlane plugin template
  * Add `rubocop` to plugin template (#4951)
  * Improved plugin default README (#4995)
* Add documentation about `itc_provider` option for deliver and pilot
* [Scan] Added option to generate the json compilation database with clang naming conventions
* Install latest release for an Xcode plugin (#4911)
* Improved the docs for hockey action (#4984)
* Fix `changelog_from_git_commits` action for CLI
* Updated `cocoapods` action to correctly update repo when using `cocoapods` 1.0
* Updated internal dependencies

KrauseFx committed:
[fastlane] Updated dependencies (#5003)

ravron committed:
Log SSH output as it is received (#4662)

* Clean up SSH script

Fixes #4504.

* Default log to true in ssh_exec

* Fix surrounding space on default assignment

KrauseFx committed:
[fastlane] Updated list of plugins (#5008)

KrauseFx committed:
Improved code style

neonichu committed:
Add CI configurations to plugin template (#5011)

- Add configuration for Circle CI
- Add configuration for Travis CI
- Ignore 'vendor/*' from Rubocop to avoid CI issues

KrauseFx committed:
Fixed OneOff actions not loading the built-in fastlane actions (#5015)

* Fixed OneOff actions not loading the built-in fastlane actions

* Fixes https://github.com/fastlane/fastlane/pull/5015/files#r65930337

KrauseFx committed:
[fastlane] Version bump (#5018)

* Fixed OneOff actions not loading the built-in fastlane actions (#5015)
* Add CI configurations to plugin template (#5011)
* Updated rubocop code style rules
* Log SSH output as it is received (#4662)

KrauseFx committed:
Reduce gem size by 70% by excluding device grid assets from .gem (#5024)

ohayon committed:
Use double quotes around options for xcrun commands (#5032)

KrauseFx committed:
[fastlane] Show error message when plugin name has _ in the name when adding a plugin (#5034)

KrauseFx committed:
Changed Helper.is_ci? to UI.interactive? (#5037)

GantMan committed:
plural agreement typo (#5041)

Commit:
Updating upload_symbols_to_sentry action to allow for bearer authenti… (#4884)

* Updating upload_symbols_to_sentry action to allow for bearer authentication token

* Code cleanup and showing warning to recommend the sentry plugin

* Added a comment about warning

* Removed another line of doc

* Fixed some syntax for hound

* More rubocop fixes

ohayon committed:
Add build setting to xcodebuild command so projects can detect snapshot during build (#5051)

Runtime detection is provided by the launch_argument in the helper.

KrauseFx committed:
Added comment on why we need word_wrap (#5040)

KrauseFx committed:
Fixed `fastlane action` table design for legacy actions (#5052)

Actions that don't use ConfigItems didn't show enough columns (missing the new default value column)

KrauseFx committed:
Updated fastlane.gemspec to include fastlane team (#5026)

jerome-mare committed:
Fix jenkins is_ci helper #1340 (#5053)

* Fix #1340 adding JENKINS_HOME to test jenkins_ci execution

* Fix #1340 adding unit test

KrauseFx committed:
Added update checker for plugins (#5039)

* Added update checker for plugins

* Addressed PR feedback

neonichu committed:
Generate empty JSON if there is no `Fastfile` (#5056)

The `--json` option should not prompt the user to setup a `Fastfile`.

KrauseFx committed:
[gym] Version bump (#5067)

* plural agreement typo (#5041)
* Improved code style
* add support for rbenv in wrap_xcodebuild/xcbuild-safe.sh (#4625)
* Enabled fastlane plugins for all the tools (#4941)
* [gym] Print out legacy information every time you run gym (#4890)
* Fix gym with path with spaces (#4721)
* Fix URL (#4604)

KrauseFx committed:
[fastlane_core] Version bump (#5072)

KrauseFx committed:
[spaceship] Version bump (#5074)

* Print proper error message when new 2step is used (#4959)
* Fix broken urls in Spaceship docs (#4952)
* [spaceship] Add .jpeg to the set of recognized media extensions
* Sort team selections (#4794)
* Removed link to RubyDoc (#4705)
* [spaceship] Updated spaceship gem description (#4684)
* [spaceship] Add link to GitHub issue for Accept-Encoding fix

dcordero committed:
push_to_git_remote: Add optional parameter 'tags' to avoid pushing tags  (#5080)

* Add tags parameter to the tests

* Include, optionally, the parameter --tags to the git command

* Update documentation of the action push_to_git_remote to describe the new parameter 'tags'

* Replace tab by spaces to make Hound happier

* Replace tab by spaces to make Hound happier

maxchuquimia committed:
UI.user_error when resign failures with non-zero exit code (#5059)

KrauseFx committed:
Added error handling when dSYM download failed (#5082)

Fixes https://github.com/fastlane/fastlane/issues/5044

neonichu committed:
Migrate `device_grid` plugin to new danger API (#5060)

* Migrate `device_grid` plugin to new danger API

* Add missing `require`

* Do not use Bundler anymore

* Make hound happy and move spec to new gem

* Update `device_grid` README

* Remove `device_grid` spec from `fastlane`

It has been moved over to the separate `device_grid` gem.

* Rename directory to `danger-device_grid`

markokeeffe committed:
Escape code signing identity

Attempting to build gym with a --codesigning_identity value that has an apostrophe (e.g. `gym --scheme "My App" --codesigning_identity "iPhone Distribution: Mark's Company (ABCDE123456)"`) will fail with the error:
```bash
sh: -c: line 0: syntax error near unexpected token `('
```

Escaping the value instead of quoting fixes the problem.

neonichu committed:
Add `danger-device_grid` to gems in `Rakefile` (#5087)

* Add `danger-device_grid` to gems in `Rakefile`

* Add rspec things

Commit:
Fix tests that are failing due to shellescape changes (#5095)

KrauseFx committed:
Removed dot in main readme (#5081)

Commit:
Update fastlane dependency on spaceship (#5096)

KrauseFx committed:
[fastlane] Version bump (#5084)

* Added error handling when dSYM download failed
* Added update checker for plugins (#5039)
* Generate empty JSON if there is no `Fastfile` (#5056)
* Updated fastlane.gemspec to include fastlane team (#5026)
* Updating `upload_symbols_to_sentry` action to allow for bearer auth (#4884)
* Changed Helper.is_ci? to UI.interactive? (#5037)
* [fastlane] Show error message when plugin name has _ in the name when adding a plugin (#5034)
* Reduce gem size by 70% by excluding device grid assets from .gem (#5024)
* Improved error handling for `download_dsyms` action
* Fixed `fastlane action` table design for legacy actions (#5052)

neonichu committed:
Merge test results and send to coveralls (#5091)

ohwutup commented:
closing this in favor of https://github.com/fastlane/fastlane/pull/5145
